ophyd/status.py

```diff
@@ -93,6 +93,9 @@ def _settle_then_run_callbacks(self, success=True):
                 self._cb = None
 
     def _finished(self, success=True, **kwargs):
+        if self.done:
+            return
```

Comment:
All should pass `done`/`success` through now, added some more docstrings as well